any_member.hpp

```diff
@@ -323,6 +331,17 @@ any_member<ClassType>& any_member<ClassType>::operator=(std::string const& s)
     return assign(s);
 }
 
+template<typename ClassType>
+any_member<ClassType>& any_member<ClassType>::operator=
+    (any_member<ClassType>&& other
+    )
+{
+    LMI_ASSERT(other.content_);
```

Comment:
I see that this is done in the assignment operator as well, but this doesn't really make sense to me. Do you understand why should it be invalid to move from an invalid/default-initialized `any_member`? If not, I'd rather leave this assert out.